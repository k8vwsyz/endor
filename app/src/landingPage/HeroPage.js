import React, {useRef, useLayoutEffect } from 'react';
import './heropage.css'
import backgroundimage from './water_background.jpg';
import Grid from '@material-ui/core/Grid'
import { makeStyles } from '@material-ui/core/styles';
import Header from '../header/Header';

const useStyles = makeStyles (() => ({
image:{
    backgroundImage: `url(${backgroundimage})`,
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
    backgroundPosition: 'center center',
    height: '100vh',
},
mainGrid:{
    height: '80vh',
    display: 'flex',
    justify:'center',
    direction: 'row',
    alignItems: 'flex-end',
},
leftText:{
    fontSize: 'min(max(18px, 4vw), 42px)',
    textAlign: 'left',
    color: 'white',
    fontFamily: 'GT America Mono,Arial,sans-serif',
},
rightText:{
    fontSize: 'min(max(18px, 4vw), 42px)',
    textAlign: 'left',
    color: 'white',
    fontFamily: 'Aeonik,Arial,sans-serif',
},
textContainer:{

}




}));

const HeroPage = (props) => {


    const {onHeightChanged} = props;
    Header.color = "black";
    const classes = useStyles();
    const heroRef = useRef(null);
    

    
    useLayoutEffect(() => {
        window.addEventListener('resize', onHeightChanged);
        onHeightChanged(heroRef.current.clientHeight);
        return () => window.removeEventListener('resize',onHeightChanged);
      }, [onHeightChanged]);

    return(

        <div className={classes.image} ref={heroRef} data-testid="heropage">
            <Grid container className={classes.mainGrid}>
                    <Grid container item xs={12} spacing={1}>
                        <Grid item sm={1}></Grid>
                        <Grid item xs={6} sm={3}>
                            <div className={classes.leftText}>
                            Vårt uppdrag:
                            </div>
                        </Grid>
                        <Grid container item xs={12} sm={6}>
                            <div className={classes.rightText}>
                                <Grid item xs={12}>Vatten är dyrbart. Vi dedikerar oss för detta syfte.</Grid>
                                <Grid item xs={12}>Vi ger människor glädje.</Grid>
                                <Grid item xs={12}>Vi är ENDOR.</Grid>
                            </div>
                        </Grid>
                    </Grid>
            </Grid>
        </div>
    )
}

export default HeroPage;