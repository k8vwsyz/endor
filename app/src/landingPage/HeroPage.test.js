import React from "react"
import { render, cleanup, act } from "@testing-library/react"
import { waitFor } from "@testing-library/dom"
import HeroPage from "./HeroPage"

afterEach(cleanup);

it("snapshot test", async () => {
    const { getByTestId, asFragment } = render(<HeroPage onHeightChanged={jest.fn(() => {})} />)
  
    await act(async () => waitFor(() => getByTestId("heropage")))
  
    expect(asFragment()).toMatchSnapshot()
  })