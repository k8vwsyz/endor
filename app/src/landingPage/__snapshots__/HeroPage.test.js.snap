// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-image-1"
    data-testid="heropage"
  >
    <div
      class="MuiGrid-root makeStyles-mainGrid-2 MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-1"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
        >
          <div
            class="makeStyles-leftText-3"
          >
            Vårt uppdrag:
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        >
          <div
            class="makeStyles-rightText-4"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              Vatten är dyrbart. Vi dedikerar oss för detta syfte.
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              Vi ger människor glädje.
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              Vi är ENDOR.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
